worker/src/r2/builder.rs

```diff
@@ -93,34 +93,44 @@ impl From<Conditional> for JsObject {
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Range {
-    OffsetWithLength { offset: u32, length: u32 },
-    OffsetWithOptionalLength { offset: u32, length: Option<u32> },
-    OptionalOffsetWithLength { offset: Option<u32>, length: u32 },
-    Suffix { suffix: u32 },
+    OffsetWithLength { offset: u64, length: u64 },
+    OffsetWithOptionalLength { offset: u64, length: Option<u64> },
+    OptionalOffsetWithLength { offset: Option<u64>, length: u64 },
+    Suffix { suffix: u64 },
+}
+
+const MAX_INT: u64 = (2 ^ 53) - 1;
```

Comment:
```suggestion
const MAX_SAFE_INTEGER: u64 = js_sys::Number::MAX_SAFE_INTEGER as u64;
```
This way, the assert semantics will be obvious.